### PR TITLE
implementing DBS(Depth Below Surface)

### PR DESCRIPTION
**Pull request by asnimansari**

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use core::str;
 
+use dbs::DbsData;
```
Comment (elpiel) [inline src/parse.rs:3]: - [ ] You need to export the DbsData in `sentences/mod.rs` like the rest of the *Data structs and you can remove this import as `sentences::*` is imported below.

File: src/sentences/dbs.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+use crate::{Error, NmeaSentence, ParseResult, SentenceType};
+
+/// DBS - Depth Below Surface
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_dbs_depth_below_surface>
+/// ```text
+///         1   2 3   4 5   6 7
+///         |   | |   | |   | |
+///  $--DBT,x.x,f,x.x,M,x.x,F*hh<CR><LF>
```
Comment (elpiel) [inline src/sentences/dbs.rs:11]:
```suggestion
///  $--DBT,x.x,f,x.x,M,x.x,F*hh<CR><LF>
/// ```
```

File: src/sentences/dbs.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+///     1. Water depth, feet
+///     2. f = feet
+///     3. Water depth, meters
+///     4. M = meters
+///     5. Water depth, Fathoms
+///     6. F = Fathoms
+///     7. Checksum
+/// In real-world sensors, sometimes not all three conversions are reported. So you might see something like $SDDBT,,f,22.5,M,,F*cs
+/// Example: $DBS,x.x,f,x.x,M,x.x,F*hh<CR><LF>
+/// ```
```
Comment (elpiel) [inline src/sentences/dbs.rs:22]:
```suggestion
///
/// In real-world sensors, sometimes not all three conversions are reported. 
/// So you might see something like `$SDDBT,,f,22.5,M,,F*cs`
///
/// Examples:
/// * `$DBS,x.x,f,x.x,M,x.x,F*hh<CR><LF>`
/// ```
```

**Comment by asnimansari** [comment]
@elpiel made the suggested changes.

File: src/sentences/dbs.rs (hunk 2 above)
Comment (asnimansari) [reply src/sentences/dbs.rs]: FIXED

File: src/sentences/dbs.rs (hunk 3 above)
Comment (asnimansari) [reply src/sentences/dbs.rs]: FIXED

File: src/sentences/dbs.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+///     1. Water depth, feet
+///     2. f = feet
+///     3. Water depth, meters
+///     4. M = meters
+///     5. Water depth, Fathoms
+///     6. F = Fathoms
+///     7. Checksum
+/// In real-world sensors, sometimes not all three conversions are reported.
+/// So you might see something like `$SDDBS,,f,22.5,M,,F*cs`
+///
```
Comment (elpiel) [inline src/sentences/dbs.rs:22]:
```suggestion
/// ```
/// In real-world sensors, sometimes not all three conversions are reported.
/// So you might see something like `$SDDBS,,f,22.5,M,,F*cs`
///
```

File: src/sentences/dbs.rs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+/// Examples:
+/// * `$DBS,x.x,f,x.x,M,x.x,F*hh<CR><LF>`
+/// ```
```
Comment (elpiel) [inline src/sentences/dbs.rs:25]:
```suggestion
/// ```
/// In real-world sensors, sometimes not all three conversions are reported.
/// So you might see something like `$SDDBS,,f,22.5,M,,F*cs`
///
/// Examples:
/// * `$DBS,x.x,f,x.x,M,x.x,F*hh<CR><LF>`
```

**Comment by elpiel** [comment]
CI checks are failing, you can check them out and fix them

**Comment by asnimansari** [comment]
> CI checks are failing; you can check them out and fix them
Yes. Fixed.
Should we think of adding a pre-commit hook for these checks?

**Comment by elpiel** [comment]
You gave me an idea 💡 

I'll see hooks and how we can set them up.

Something, though, you need to run clippy and rustfmt with nightly but build your code using stable rust.

Imo This is out of scope for this pr

**Comment by asnimansari** [comment]
@elpiel could you help me with the remaining checks?
What needs to be changed?

**Comment by elpiel** [comment]
It's ready to be merged.
Could you just add an example sentence here:
https://github.com/AeroRust/nmea/blob/main/tests%2Fall_supported_messages.rs

**Comment by asnimansari** [comment]
> It's ready to be merged. Could you just add an example sentence here: https://github.com/AeroRust/nmea/blob/main/tests%2Fall_supported_messages.rs

@elpiel done.

File: tests/all_supported_messages.rs [hunk 6]
```diff
@@ -54,7 +54,9 @@ fn test_all_supported_messages() {
         // ZTG
         (SentenceType::ZTG, "$GPZTG,145832.12,042359.17,WPT*24"),
         // DPT (Depth of Water)
-        (SentenceType::DPT, "$SDDPT,17.9,0.5*6D")
+        (SentenceType::DPT, "$SDDPT,17.9,0.5*6D"),
+        // DPT
+        (SentenceType::DPT, "$SDDBS,12.3,f,3.75,M,2.05,F*37"),
```
Comment (elpiel) [inline tests/all_supported_messages.rs:59]:
```suggestion
        // DBS
        (SentenceType::DBS, "$SDDBS,12.3,f,3.75,M,2.05,F*37"),
```

**Comment by asnimansari** [comment]
cool
@elpiel how do we merge this?

**Comment by elpiel** [comment]
Thank you once again for opening the pr!

I approved the run and just had to wait for it to finish 😋

**Comment by asnimansari** [comment]
Cool. Thanks. 
@elpiel

**Pull request merged**